### PR TITLE
middleware/user-autenticate function validateJWT added, verify if tok…

### PR DESCRIPTION
**Pull request by sebasam**
### Summary

- middlewares/user-autenticate.js and function validateJWT if token belongs to any user.

### CODE REVIEW

File: middlewares/user-autenticate.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    const token = req.header('x-token');
+    if(!token){
+        return res.status(403).json({
+            msg: 'Token is not valid!'
+        })
+    }
+    try {
+        const { id, firstName } = jwt.verify(token, process.env.JWT_SECRET);
+        req.id = id;
+        req.firstName = firstName;
```
Comment (maxxCeballos) [inline middlewares/user-autenticate.js]: Estaria bueno sacar todos los datos del usuario para poder validar roles y demas.

**Review by maxxCeballos — CHANGES_REQUESTED** [review]

**Comment by sebasam** [comment]
### Summary

- User info has been deleted from middleware

File: middlewares/user-autenticate.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+const validateJWT = ( req = request, res = response, next ) => {
+    const token = req.header('x-token');
+    if(!token){
+        return res.status(403).json({
+            msg: 'Token is not valid!'
+        })
+    }
+    try {
+        jwt.verify(token, process.env.JWT_SECRET);
```
Comment (maxxCeballos) [inline middlewares/user-autenticate.js]: Seba no me expresé bien, me referia a obtener el user completo del token y agregarlo como atributo de req.

**Review by maxxCeballos — CHANGES_REQUESTED** [review]

**Comment by sebasam** [comment]
### Summary

- User data added like req attribute from token, excuseme for that, i hadnt understand the context for the first change request jeje

**Comment by sebasam** [comment]
### Summary

- Filename changed to user-authenticate.js
- bearer token logic added and user data received

Sorry, i hadnt undestand the problem context

**Review by maxxCeballos — APPROVED** [review]

**Pull request merged**